core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/util/SignerConstant.java

```diff
@@ -13,12 +13,12 @@
  * permissions and limitations under the License.
  */
 
-package software.amazon.awssdk.http.auth.internal.util;
+package software.amazon.awssdk.http.auth.aws.util;
 
 import java.time.Duration;
-import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
 
-@SdkInternalApi
+@SdkProtectedApi
 public final class SignerConstant {
```

Comment:
Any way to avoid this needing to be protected? Or can we reduce the number of things it contains to strictly what is needed to be protected?

Reply:
I guess we can make things that aren't used cross-module package-private?

Reply:
I guess this is a good topic - do we want a constants file or do we want to simply define constants in the file in which they are used? 

I prefer the latter, even when there is *some* duplication across files.

Reply:
In general if a string value needs to be used in multiple places, defining it somewhere are referencing that is better. This is easier answer when it's all in one module (can be SdkInternalApi), but when it is becoming cross-module SdkProtectedApi, I think we need to evaluate case-by-case. I'm finding it hard to tell right now, with the scope of this PR not including sigv4a/streaming I don't see the cross-module use. So I think it might be better to keep this class SdkInternalApi for this PR, and if it is changing to support the subsequent PR, discussing it there will be more focused discussion. Just my 2 cents.